src/mrsimulator/method/lib/base.py

```diff
@@ -34,9 +35,10 @@ def check_for_method_name(cls, v, *, values, **kwargs):
 
     @validator("rotor_frequency", pre=True, always=True)
     def check_rotor_frequency(cls, v, *, values, **kwargs):
+        v = 1.0e12 if np.isinf(v) else v  # Allow np.inf for named methods
         a = (
             True if "name" not in values else values["name"] == "SSB2D",
```

Comment:
Since `name` is an attribute of the method, is this statement  
```py
if "name" not in values
```
ever true? Might as well remove it then!

Reply:
Statement has been changed to check for

1. SSB2D method
2. 1e12 Hz (equivalent string and float)
3. 1D method

If any of these are true, then the rotor frequency is valid.